index.js

```diff
@@ -13,16 +14,22 @@ module.exports.file = function file$promise() {
 
 module.exports.withFile = function withFile(fn) {
   var cleanup;
-  return module.exports.file.apply(tmp).then(function context(o) { 
+
+  var params = Array.prototype.slice.call(arguments);
+  params.shift();
```

Comment:
`.shift()` remove the first entry from the array, in this case the `fn`. We can use instead `.slice(1)` if it's cleaner, and now that I think about it, we are already using `.slice()`, hum...

Yes, we could use `Array.from()`, it's just that I always forgot about it, I'm too much used to the `.slice()` hack :-P